tests/integration/check-your-answers/file-upload-site-details.test.js

```diff
@@ -0,0 +1,154 @@
+import { JSDOM } from 'jsdom'
+import { routes } from '~/src/server/common/constants/routes.js'
+import { statusCodes } from '~/src/server/common/constants/status-codes.js'
+import * as authRequests from '~/src/server/common/helpers/authenticated-requests.js'
+import * as cacheUtils from '~/src/server/common/helpers/session-cache/utils.js'
+import { createServer } from '~/src/server/index.js'
+import { testScenarios } from './fixtures.js'
+
+jest.mock('~/src/server/common/helpers/session-cache/utils.js')
+jest.mock('~/src/server/common/helpers/authenticated-requests.js')
+
+describe('Check Your Answers - File Upload Site Details', () => {
+  let server
+
+  beforeAll(async () => {
+    server = await createServer()
+    await server.initialize()
+  })
+
+  afterAll(async () => {
+    await server.stop()
+  })
+
+  beforeEach(() => {
+    jest.resetAllMocks()
+    jest
+      .spyOn(cacheUtils, 'setExemptionCache')
+      .mockImplementation(() => undefined)
+  })
+
+  test.each(testScenarios)(
+    '$name - validates every element on the page',
+    async ({ exemption, expectedPageContent }) => {
+      expect.hasAssertions()
+
+      const document = await getPageDocument(exemption)
+
+      validatePageStructure(document, expectedPageContent)
+      validateAllSummaryCardsExist(document, expectedPageContent)
+      validateProjectDetails(document, expectedPageContent)
+      validateActivityDates(document, expectedPageContent)
+      validateActivityDetails(document, expectedPageContent)
+      validateSiteDetails(document, expectedPageContent)
+      validatePublicRegister(document, expectedPageContent)
+      validateSubmissionSection(document, expectedPageContent)
+    }
+  )
+
+  const getPageDocument = async (exemption) => {
+    jest.spyOn(cacheUtils, 'getExemptionCache').mockReturnValue(exemption)
+    jest.spyOn(authRequests, 'authenticatedGetRequest').mockResolvedValue({
+      payload: { value: { taskList: { id: exemption.id } } }
+    })
+
+    const response = await server.inject({
+      method: 'GET',
+      url: routes.CHECK_YOUR_ANSWERS
+    })
+
+    expect(response.statusCode).toBe(statusCodes.ok)
+    return new JSDOM(response.result).window.document
+  }
+
+  const validatePageStructure = (document, expected) => {
+    const heading = document.querySelector('#check-your-answers-heading')
+    expect(heading.textContent.trim()).toBe(expected.pageTitle)
+
+    const backLink = document.querySelector('.govuk-back-link')
+    expect(backLink.textContent.trim()).toBe(expected.backLinkText)
+  }
```

Comment:
DOM testing library `getByRole('link', { name: expected.backLinkText})` is more compact and readable, and checks it's a link as well as the link text?